controllers/mysqlcluster_controller.go

```diff
@@ -1862,6 +1858,51 @@ func (r *MySQLClusterReconciler) finalizeV1(ctx context.Context, cluster *mocov1
 	return nil
 }
 
+func (r *MySQLClusterReconciler) UpdateStatus(ctx context.Context, req ctrl.Request, cluster *mocov1beta2.MySQLCluster) error {
+	log := crlog.FromContext(ctx)
+
+	var sts appsv1.StatefulSet
+	err := r.Get(ctx, client.ObjectKey{Namespace: cluster.Namespace, Name: cluster.PrefixedName()}, &sts)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("failed to get StatefulSet %s/%s: %w", cluster.Namespace, cluster.PrefixedName(), err)
+	}
+
+	needUpdate := false
+	updated := metav1.ConditionFalse
+	reason := "UpdateInProgress"
+	if sts.Status.AvailableReplicas != 0 && sts.Status.CurrentRevision == sts.Status.UpdateRevision && sts.Generation == sts.Status.ObservedGeneration {
+		updated = metav1.ConditionTrue
+		reason = "UpdateCompleted"
+	}
+	if cluster.Status.ReconcileInfo.Generation != cluster.Generation {
+		cluster.Status.ReconcileInfo.Generation = cluster.Generation
+		cluster.Status.ReconcileInfo.ReconcileVersion = 1
+		needUpdate = true
+	}
+	conditionUpdated := meta.FindStatusCondition(cluster.Status.Conditions, mocov1beta2.ConditionUpdated)
+	// if current status and new status are different, update status
+	if conditionUpdated == nil || conditionUpdated.Status != updated {
+		needUpdate = true
+	}
+	if needUpdate {
+		cond := metav1.Condition{
+			Type:               mocov1beta2.ConditionUpdated,
+			Status:             updated,
+			Reason:             reason,
+			Message:            "the current state is " + reason,
+			LastTransitionTime: metav1.Now(),
```

Comment:
LastTransitionTime would be set automatically in SetStatusCondition.
https://github.com/kubernetes/apimachinery/blob/5cb236977966cfc829e3b94b57fa69b44e0a95bc/pkg/api/meta/conditions.go#L30

Reply:
Can you set `ObservedGeneration`?

Reply:
>Can you set ObservedGeneration?

What I have to set in this value is `cluster.Generation` at the point of reconciliation, is it right?

Reply:
Yes.